en/components/general/wpf_to_angular_guide/structural_directives.md

```diff
@@ -0,0 +1,146 @@
+---
+
+## Structural Directives in Angular
+
+As WPF developers, anytime we want to add or remove an element from the visual tree we have to jump into some code-behind and write some C# or we can use a combination of binding and a visibility converter, which again requires some custom logic and static resources. That is somewhat OK and we have always done it in WPF, but Angular makes it so much easier.
+Manipulating your view in an Angular application /adding and removing elements/, is achieved with the so-called "structural directives". In this topic, we are going to demonstrate the following three directives - `ngIf`, `ngSwitch` and `ngFor`. As one can tell from their names, each of these can be compared to a C# structure. The `ngIf` is the same thing as an "if-else" C# code block, the `ngSwitch` is the same thing as the C# switch-case statement and, lastly, the `ngFor` is the exact same thing as a C# "for-loop".
+
+### `ngIf` Directive
+Now, let’s explore each of these directives, starting with the `ngIf`. This directive allows us to show or hide elements based on a boolean condition. We will start by creating a "div" element with an "h2" tag containing a name.
+```html
+<div>
+    <h2>John</h2>
+</div>
+```
+If we save this our browser will render the name John. However, let’s say we have some type of boolean expression that we want to base the condition of the visibility of this "h2" tag on. For example, we are going to add a property called "isFirstName" and set it to false. In order to tell our div to be rendered when isFirstName equals true, we should use the following syntax *ngIf = "isFirstName".
+```typescript
+  public isFirstName = false;
+```
+``` html
+<div *ngIf="isFirstName">
+  <h2>John</h2>
+</div>
+```
+Once we save the files and because isFirstName is false, we will see that the name is no longer rendered on the browser. However, if we were to update this to be true, save that value so the condition resolves as true then the "John" first name will be rendered in the browser. If we set this first name back to false and we'll notice that the first name is no longer rendered in our browser, instead it's empty. That's the default behavior of the if statement - if the expression is true we render the provided template otherwise it's empty.
+If we were to achieve the same behavior with WPF, we would need to use a visibility converter. The code would look similar to the following:
+```cs
+public bool IsFirstName { get; set; }
+public Sample()
+  {
+      InitializeComponent();
+      this.DataContext = this;
+      this.IsFirstName = true;
+  }
+```
+```html
+<UserControl.Resources>
+  <BooleanToVisibilityConverter x:Key="VisibleIfTrueConverter" />
+</UserControl.Resources>
+<Grid>
+  <Label Visibility="{Binding Path=IsFirstName, Converter={StaticResource VisibleIfTrueConverter}}">John</Label>
+</Grid>
+```
+A lot easier and more straightforward it is in Angular.
```

Comment:
A lot easier and more straightforward it is in Angular. -> In Angular, it is a lot easier and more straightforward.

Reply:
Addressed.